cmd/otel-allocator/Dockerfile

```diff
@@ -1,26 +1,34 @@
 # Build the target allocator binary
 FROM golang:1.19 as builder
 
-WORKDIR /app
-
-# Copy go mod and sum files
-COPY go.mod go.sum ./
-
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
-COPY . .
+# Copy the go source
```

Comment:
Does it make senes to keep `COPY . .`? Unused files are not copied in the next stage anyway?

Reply:
I'm not sure about that files you're referring. The problem with that is that other files that you don't want to be copied, will be copied too (and if those files change, the Docker cache will be invalidated at that layer).

But I don't have a strong opinion: I just tried to follow the same approach than in the `Dockerfile` from the root folder. If you think we should follow the `COPY . .` approach, maybe we should create a new task to address that.

Reply:
I don't have a strong opinion either. My thought was, since the overhead of files should normally be quite limited and it is the build stage, it should not be so bad to just copy the complete folder content. At least this way you won't forget to copy something / need to touch the Dockerfile again.
As for caching, I don't think it makes much difference. Since it is only a few text files that are copied.

Reply:
since this is only the build stage I think it best to copy the entire context.  That way we don't need to remember to update this file if we add new packages or `.go` files in the root.